### ISSUE TITLE
StubBroker vs RedisBroker+redislite

### ISSUE THREAD

**Issue by rpkilby** [Bogdanp/dramatiq#58]
During testing, what are the practical differences between the `StubBroker` and the `RedisBroker`+redislite?

**Comment by Bogdanp** [comment]
I've never used `redislite` (though it looks pretty cool!) so I wouldn't really know. The biggest difference would be that the stub broker doesn't have any dependencies, but they should both behave the same. I don't think I'd want to replace the stub broker with that combination, however, because it doesn't feel right to force someone who, for example, only uses the RMQ broker to also install Redis if he/she doesn't need it.

**Comment by rpkilby** [comment]
Definitely agree - the `StubBroker` is a suitable testing alternative for RMQ users. Mostly just wanted to check if the StubBroker had any behavior differences or additional testing features that I would otherwise miss out on by using redislite.

**Comment by rpkilby** [comment]
Ah - the `RedisBroker` does not have the `flush` and `flush_all` methods.

**Comment by Bogdanp** [comment]
Right, but it doesn't really need them since you can just flush Redis directly.

**Comment by Bogdanp** [comment]
Makes sense. I opened https://github.com/Bogdanp/dramatiq/issues/62 to track that as an enhancement.
